index.js

```diff
@@ -17,11 +18,24 @@ CoffeeScriptFilter.prototype.targetExtension = 'js'
 CoffeeScriptFilter.prototype.processString = function (string, srcFile) {
   var coffeeScriptOptions = {
     bare: this.bare,
-    literate: coffeeScript.helpers.isLiterate(srcFile)
+    literate: coffeeScript.helpers.isLiterate(srcFile),
+    sourceMap: !!this.sourceMap,
+
+    sourceFiles: [ srcFile ],
```

Comment:
Thanks for the PR! Given an input file `dir/foo.coffee` that gets compiled into `dir/foo.js`, this `sourceFiles` option will result in the following source map output:

```
  "sources": [
    "dir/foo.coffee"
  ],
```

But "sources" are URLs relative to `dir/foo.js`. Thus if you load `/dir/foo.js` with a `<script>` tag, Chrome inspector shows the CoffeeScript source as `dir/dir/foo.coffee` in its sources panel.

Using `sourceFiles: [ path.basename(srcFile) ]` here is a possible workaround, but it seems that this should really be fixed on CoffeeScript's end: https://github.com/jashkenas/coffeescript/issues/3296
I'd be OK using the `basename` workaround though if you think that's reasonable.